docs/tutorials/mastering.md

```diff
@@ -0,0 +1,618 @@
+
+```console
+juju offer control-plane.amf:fiveg-n2
+```
+
+Configure the NMS to be served as `control-plane.mgmt`:
+
+```console
+juju config traefik-k8s external_hostname=control-plane.mgmt
+```
+
+### Checkpoint
+
+You should be able to see the AMF External load balancer service in Kubernetes. Log into the control plane cluster and execute the following command:
+
+```console
+microk8s.kubectl get services -A | grep LoadBalancer
+```
+
+This will show output similar to the following, indicating:
+- The AMF is exposed on 10.201.0.100 SCTP port 38412
+- The NMS is exposed on 10.201.0.101 TCP ports 80 and 443
+
+These IP addresses came from MetalLB and were configured in the bootstrap step.
+
+```
+control-plane    amf-external  LoadBalancer  10.152.183.179  10.201.0.100   38412:30408/SCTP
+control-plane    traefik-k8s   LoadBalancer  10.152.183.28   10.201.0.101   80:32349/TCP,443:31925/TCP
+```
+
+## 3. Deploy SD-Core User Plane
+
+Create a Juju model to represent the User Plane, using the cloud `user-plane-cluster`, which was created in the Bootstrap step. The following parameters can be set:
+
+- `access-gateway-ip`: this is the IP address of the gateway that knows how to route traffic from the UPF towards the gNB subnet
+- `access-interface`: the name of the MACVLAN interface on the Kubernetes host cluster to bridge to the `access` subnet
+- `access-ip`: the IP address for the UPF to use on the `access` subnet
+- `core-gateway-ip`: this is the IP address of the gateway that knows how to route traffic from the UPF towards the internet
+- `core-interface`: the name of the MACVLAN interface on the Kubernetes host cluster to bridge to the `core` subnet
+- `core-ip`: the IP address for the UPF to use on the `core` subnet
+- `gnb-subnet`: the subnet CIDR where the gNB radios are reachable.
+
+```{note}
+The `access-gateway-ip` is expected to forward the packets from the `access-interface` to the `gnb-subnet`
+```
+
+```console
+cat << EOF > upf-overlay.yaml
+applications:
+  upf:
+    options:
+      access-gateway-ip: 10.202.0.1
+      access-interface: access
+      access-ip: 10.202.0.10/24
+      core-gateway-ip: 10.203.0.1
+      core-interface: core
+      core-ip: 10.203.0.10/24
+      gnb-subnet: 10.204.0.0/24
+      external-upf-hostname: upf.mgmt
+EOF
+```
+
+Next, we create a Juju model to represent the User Plane, using the cloud `user-plane-cluster`, which was created in the Bootstrap step.
+
+```console
+juju add-model user-plane user-plane-cluster
+```
+
+Deploy user plane bundle:
+
+```console
+juju deploy sdcore-user-plane --trust --channel=edge --overlay upf-overlay.yaml
+```
+
+Now expose the UPF service offering with Juju.
+
+```console
+juju offer upf:fiveg_n4
+```
+
+### Checkpoint
+
+You should be able to see the UPF External load balancer service in Kubernetes. Log into the user plane cluster and execute the following command:
+
+```console
+microk8s.kubectl get services -A | grep LoadBalancer
```

Comment:
We never mention adding user to `snap_microk8s` group in this host, so this command will fail.